lib/functions/worker.sh

```diff
@@ -249,6 +249,14 @@ function LintCodebase() {
           ${LINTER_COMMAND} "${FILE_NAME}" | tee /dev/tty2 2>&1
           exit "${PIPESTATUS[0]}"
         )
+      #########################################################
+      # Corner case for CFN_NAG as path is passed inside a flag #
+      #########################################################
+      elif [[ ${FILE_TYPE} == "YAML" ]] || [[ ${FILE_TYPE} == "JSON" ]]; then
```

Comment:
So CONFUSINGLY, `${FILE_TYPE}` is not really a 'file type' (as the name suggests), but actually just a language defined in your `LANGUAGE_ARRAY` (inside linter.sh):

- [functions/worker.sh:18](https://github.com/23andMe/super-linter/blob/master/lib/functions/worker.sh#L18) - `${FILE_TYPE}` is the first arg to `LintCodebase`
- [linter.sh:953](https://github.com/23andMe/super-linter/blob/master/lib/linter.sh#L953) - Where `LintCodebase` is called, and its first arg is `${LANGUAGE}`
- [linter.sh:906](https://github.com/23andMe/super-linter/blob/master/lib/linter.sh#L906) - ...and `${LANGUAGE}` is just one of the elements of `LANGUAGE_ARRAY`
- [linter.sh:196](https://github.com/23andMe/super-linter/blob/master/lib/linter.sh#L196) - this was where you defined your languages inside `LANGUAGE_ARRAY`

So in short, for this line you can be more specific for cfn_nag and just use the new `CLOUDFORMATION_CFN_NAG` value you've added to `LANGUAGE_ARRAY`:
```
      elif [[ ${FILE_TYPE} == "CLOUDFORMATION_CFN_NAG" ]]; then
```

Reply:
Oh thank you Sarah, I tried to debug this whole day yesterday !